File: mobile/lib/models/comment.dart
class Comment {
  final int id;
  final int jobId;
  final String name;
  final String comment;
  final int? parentId;
  final String? createdAt;

  Comment({
    required this.id,
    required this.jobId,
    required this.name,
    required this.comment,
    this.parentId,
    this.createdAt,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['ID'] ?? json['id'] ?? 0,
      jobId: json['JobID'] ?? json['job_id'] ?? 0,
      name: json['Name'] ?? json['name'] ?? '',
      comment: json['Comment'] ?? json['comment'] ?? '',
      parentId: json['ParentID'] ?? json['parent_id'],
      createdAt: json['CreatedAt'] ?? json['created_at'],
    );
  }
}

File: backend/models/comment.go
package models

import "gorm.io/gorm"

type Comment struct {
	gorm.Model
	JobID    uint   `json:"job_id" gorm:"not null;index"`
	UserID   uint   `json:"user_id"`
	Name     string `json:"name" gorm:"not null"`
	Text     string `json:"text" gorm:"type:text;not null"`
	ParentID *uint  `json:"parent_id"`
}
